airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/jwt.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import InitVar, dataclass
+from datetime import datetime
+from typing import Any, Mapping, Optional, Union
+
+import jwt
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import DeclarativeAuthenticator
+from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+
+
+@dataclass
+class JwtAuthenticator(DeclarativeAuthenticator):
+
+    config: Mapping[str, Any]
+    parameters: InitVar[Mapping[str, Any]]
+    secret_key: Union[InterpolatedString, str]
+    algorithm: Union[InterpolatedString, str]
+    token_duration: Union[InterpolatedString, str] = 1200
+    kid: Union[InterpolatedString, str] = None
+    typ: Union[InterpolatedString, str] = "JWT"
+    iss: Union[InterpolatedString, str] = None
+    sub: Union[InterpolatedString, str] = None
+    aud: Union[InterpolatedString, str] = None
+    cty: Union[InterpolatedString, str] = None
+    additional_jwt_headers: Mapping[str, Any] = None
+    additional_jwt_payload: Mapping[str, Any] = None
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        super().__init__()
+        self._algorithm = InterpolatedString.create(self.algorithm, parameters=parameters)
+        self._secret_key = InterpolatedString.create(self.secret_key, parameters=parameters)
+        self._kid = InterpolatedString.create(self.kid, parameters=parameters)
+        self._typ = InterpolatedString.create(self.typ, parameters=parameters)
+        self._iss = InterpolatedString.create(self.iss, parameters=parameters)
+        self._sub = InterpolatedString.create(self.sub, parameters=parameters)
+        self._aud = InterpolatedString.create(self.aud, parameters=parameters)
+        self._cty = InterpolatedString.create(self.cty, parameters=parameters)
+        self._token_duration = self.token_duration
+        self._additional_jwt_headers = InterpolatedMapping(self.additional_jwt_headers or {}, parameters=parameters)
+        self._additional_jwt_payload = InterpolatedMapping(self.additional_jwt_payload or {}, parameters=parameters)
+
+    def _get_jwt_headers(self) -> Mapping[str, Any]:
+        headers = {}
+        headers.update(self._additional_jwt_headers.eval(self.config))
+        if self._kid:
+            headers["kid"] = f"{self._kid.eval(self.config)}"
```

Comment:
should we fail if "kid" or other required params was also defined in the additional jwt headers?

Reply:
Good point. Added a check for both `jwt_headers` and `jwt_payload`.